backend/src/Designer/Controllers/DatamodelsController.cs

```diff
@@ -48,12 +48,19 @@ public DatamodelsController(ISchemaModelService schemaModelService)
         [Route("datamodel")]
         public async Task<ActionResult<string>> Get([FromRoute] string org, [FromRoute] string repository, [FromQuery] string modelPath)
         {
-            var decodedPath = Uri.UnescapeDataString(modelPath);
+            try
+            {
+                var decodedPath = Uri.UnescapeDataString(modelPath);
 
-            var developer = AuthenticationHelper.GetDeveloperUserName(HttpContext);
-            var json = await _schemaModelService.GetSchema(org, repository, developer, decodedPath);
+                var developer = AuthenticationHelper.GetDeveloperUserName(HttpContext);
+                var json = await _schemaModelService.GetSchema(org, repository, developer, decodedPath);
 
-            return Ok(json);
+                return Ok(json);
+            }
+            catch (IOException)
```

Comment:
I suggest replacing all `IOException` with `FileNotFoundException`, as `IOException` could catch errors unrelated to a non existing file (e.g. permission issues).

```suggestion
            catch (FileNotFoundException)
```